File: src/Bloemert.Data.Entity.Admin/Entity/Log.cs
using Bloemert.Data.Core;
using Bloemert.Data.Entity.Auth;
using Bloemert.Data.Entity.Auth.Entity;
using System;
using System.Xml;

namespace Bloemert.Data.Entity.Admin.Entity
{
	public class Log : BaseEntity
	{
		public virtual string Message { get; set; }

   public virtual string Level { get; set; }

		public virtual DateTimeOffset TimeStamp { get; set; }

		public virtual string Exception { get; set; }

		//public virtual XmlDocument Properties { get; set; }

		public virtual string LogEvent { get; set; }

	}
}

File: src/Einstein.WebUI/ClientApp/components/admin/models/Log.js

export class Log {
	constructor({
		seqno = 0,

		status = {
			valid: true,
			added: false,
			modified: false,
			removed: false,
			loaded: false
		},

		id = -1,

		message = '',
		level = '',
		timeStamp = Date.now(),
		exception = '',
		properties = '',
		logEvent = ''
		} = {}) {

		this.seqno = seqno;
		this.status = status;

		this.id = id;

		this.message = message;
		this.level = level;

		this.timeStamp = timeStamp;
		this.exception = exception;
		this.properties = properties;

		this.logEvent = logEvent;
	}
}

export function createLog(data) {
	return Object.freeze(new Log(data));
}